src/database.c

```diff
@@ -493,3 +555,159 @@ _add_to_db(ProfMessage* message, char* type, const Jid* const from_jid, const Ji
         }
     }
 }
+
+static int
+_get_db_version(void)
+{
+    int current_version = -1;
+    const char* query = "SELECT `version` FROM `DbVersion` LIMIT 1";
+    sqlite3_stmt* statement;
+
+    if (sqlite3_prepare_v2(g_chatlog_database, query, -1, &statement, NULL) == SQLITE_OK) {
+        if (sqlite3_step(statement) == SQLITE_ROW) {
+            current_version = sqlite3_column_int(statement, 0);
+        }
+        sqlite3_finalize(statement);
+    }
+    return current_version;
+}
+
+/**
+ * @brief Migration to V2 that introduces new fields and reorganizes message storage.
+ * `replace_id` -> `replaces_stanza_id` stanza ID of the replaced message
+ * `replaces_db_id` db ID of the replaced message
+ * `original_message` original message before replacement
+ * `message` now is used to store visual message (replaced by LMC)
+ *
+ * @return gboolean with migration success status.
+ */
+static gboolean
+_migrate_to_v2(void)
+{
+    char* err_msg;
+
+    // Make V2 table
+    char* query = "CREATE TABLE `ChatLogsV2` (`id` INTEGER PRIMARY KEY AUTOINCREMENT, `from_jid` TEXT NOT NULL, `to_jid` TEXT NOT NULL, `from_resource` TEXT, `to_resource` TEXT, `message` TEXT, `timestamp` TEXT, `type` TEXT, `stanza_id` TEXT, `archive_id` TEXT, `encryption` TEXT, `marked_read` INTEGER, `original_message` TEXT, `replaces_db_id` INTEGER, `replaces_stanza_id` TEXT)";
```

Comment:
It does not make sense to call it ChatLogsV2. We want the version only in the db_version. Not in the name of the table. The table should be called ChatLogs.

Reply:
> It does not make sense to call it ChatLogsV2. We want the version only in the db_version. Not in the name of the table. The table should be called ChatLogs in the end. During transaction could have the V2 there or something like "transformation" or something. But in the end it should be ChatLogs again.

Well, it's renamed later to "ChatLogs" (once all the fields are transformed), so I guess it's resolved.